zebra-chain/src/orchard/tests/vectors.rs

```diff
@@ -0,0 +1,47 @@
+use halo2::arithmetic::FieldExt;
+use halo2::pasta::pallas;
+
+use crate::orchard::tests::test_vectors;
+use crate::orchard::tree::*;
+
+#[test]
+fn empty_roots() {
+    zebra_test::init();
+
+    for i in 0..EMPTY_ROOTS.len() {
+        assert_eq!(
+            EMPTY_ROOTS[i].to_bytes(),
+            // The test vector is in reversed order.
+            test_vectors::EMPTY_ROOTS[MERKLE_DEPTH - i]
+        );
+    }
+}
+
+#[test]
+fn incremental_roots() {
+    zebra_test::init();
+
+    let mut leaves = vec![];
+
+    let mut incremental_tree = NoteCommitmentTree::default();
+
+    for (i, commitment_set) in test_vectors::COMMITMENTS.iter().enumerate() {
+        for cm_x in commitment_set.iter() {
+            let cm_u = pallas::Base::from_bytes(&cm_x).unwrap();
+
+            leaves.push(cm_u);
+
+            let _ = incremental_tree.append(cm_u);
```

Comment:
```suggestion
        for cm_x_bytes in commitment_set.iter() {
            let cm_x = pallas::Base::from_bytes(&cm_x_bytew).unwrap();

            leaves.push(cm_x);

            let _ = incremental_tree.append(cm_x);
```

cm_x refers to the x-coordinate of the Orchard note commitment, which is an affine point on the Pallas curve in short Weierstrass form. 

cm_u refers to the u-coordinate of the Sapling note commitment, which is an affine point on the Jubjub  curve in Edwards form, thus is inaccurate here

Reply:
You're right! It was a leftover from copying code from the Sapling test :sweat: Fixed in 97ec26174